### PR TITLE
[DependencyInjection] Add `urlencode` function to `EnvVarProcessor`

### LINKED ISSUE #19147: [DependencyInjection] Add `urlencode` function to `EnvVarProcessor`
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#52369
| PR author(s) | @crtl
| Merged in    | 7.1

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.1 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alamirault**
Fix #19147

### CODE REVIEW

**Comment by alexandre-daubois** [comment]
This should target 7.1 :slightly_smiling_face:

File: configuration/env_var_processors.rst [hunk 1]
```diff
@@ -824,6 +824,56 @@ Symfony provides the following env var processors:
...
+            use Symfony\Config\FrameworkConfig;
+
+            return static function (ContainerBuilder $container): void {
+                $container->setParameter('env(DATABASE_URL)', 'mysql://db_user:foo@b$r@127.0.0.1:3306/db_name');
+                $container->setParameter('encoded_database_url', '%env(urlencode:DATABASE_URL)%');
+            };
+
+    .. versionadded:: 6.4
+
+        The ``env(urlencode:...)`` env var processor was introduced in Symfony 6.4.
```
Comment (OskarStark) [inline configuration/env_var_processors.rst]:
```suggestion
        The ``env(urlencode:...)`` env var processor was introduced in Symfony 7.1.
```

**Comment by alamirault** [comment]
I read code target branch :man_facepalming: 

Rebased on 7.1 now

**Comment by alexandre-daubois** [comment]
Should we mention https://www.php.net/manual/en/function.urlencode.php somewhere? This may help people using it and bring additional doc to this feature :slightly_smiling_face:

**Comment by javiereguiluz** [comment]
Thanks Antoine!

Alex, I agree with you. We added the mention to PHP's `urlencode()` function while merging.

**Comment by alamirault** [comment]
Thank you both :+1:

**Pull request merged**